<launch>
  <!--rgbd导航+轮式里程机-->

  <!-- Choose visualization -->
  <arg name="rviz" default="true" />
  <arg name="rtabmapviz" default="true" />

  <!--$ Localization-only mode -->
  <arg name="localization" default="true"/>

  <!-- Corresponding config files -->
  <arg name="rtabmapviz_cfg" default="~/.ros/rtabmap_gui.ini" />
  <arg name="rviz_cfg" default="$(find jhr_navigation)/rviz/rtabmap_nav.rviz" />

  <arg name="frame_id" default="base_footprint"/>
  <!-- Fixed frame id, you may set "base_link" or "base_footprint" if they are published -->
  <arg name="database_path" default="/home/jhr/.ros/default.db"/>

  <arg name="rtabmap_args" default=""/>
  <!-- delete_db_on_start, udebug -->
  <arg name="launch_prefix" default=""/>
  <!-- for debugging purpose, it fills launch-prefix tag of the nodes -->
  <arg name="approx_sync" default="true"/>
  <!-- if timestamps of the input topics are not synchronized -->

  <arg name="rgb_topic" default="/camera/infra1/image_rect_raw" />
  <arg name="depth_registered_topic" default="/camera/depth/image_rect_raw" />
  <arg name="camera_info_topic" default="/camera/infra1/camera_info" />
  <arg name="compressed" default="false"/>

  <arg name="subscribe_scan" default="true"/>
  <arg name="scan_topic" default="/scan"/>

  <arg name="subscribe_scan_cloud" default="false"/>
  <arg name="scan_cloud_topic" default="/scan_cloud"/>
  <!--不适用rtabmap的里程计-->
  <arg name="visual_odometry" default="false"/>
  <arg name="icp_odometry" default="false"/>

  <!--￥ 使用激光或视觉里程机-->
  <!-- <arg name="odom_topic" default="rtabmap_odom"/>          -->
  <!-- <arg name="odom_frame_id" default=""/> -->

  <!-- hector_odom-->
  <!-- <arg name="odom_topic" default="/hector/slam_out_pose"/> -->
  <!-- <arg name="odom_frame_id" default="hector_map"/> --> 

  <!--wheel_odom-->
  <arg name="odom_topic" default="/wheel_odom"/>
  <arg name="odom_frame_id" default="wheel_odom_link"/> 

  <!--robot_pose_ekf-->
  <!-- <arg name="odom_topic" default="/robot_pose_ekf/odom_combined"/> -->
  <!-- <arg name="odom_frame_id" default="/odom_combined"/>  -->

  <!--rf2o_odom-->
  <!-- <arg name="odom_topic" default="/rf2o_odom"/> -->
  <!-- <arg name="odom_frame_id" default="rf2o_odom_link"/> -->

  <arg name="namespace" default="rtabmap"/>
  <arg name="wait_for_transform" default="1"/>

  <include file="$(find jhr_navigation)/launch/offical/rtabmap_default.launch">
    <arg name="rtabmapviz" value="$(arg rtabmapviz)" />
    <arg name="rviz" value="$(arg rviz)" />
    <arg name="localization" value="$(arg localization)"/>
    <arg name="gui_cfg" value="$(arg rtabmapviz_cfg)" />
    <arg name="rviz_cfg" value="$(arg rviz_cfg)" />

    <arg name="frame_id" value="$(arg frame_id)"/>
    <arg name="namespace" value="$(arg namespace)"/>
    <arg name="database_path" value="$(arg database_path)"/>
    <arg name="wait_for_transform" value="$(arg wait_for_transform)"/>
    <arg name="rtabmap_args" value="$(arg rtabmap_args)"/>
    <arg name="launch_prefix" value="$(arg launch_prefix)"/>
    <arg name="approx_sync" value="$(arg approx_sync)"/>

    <arg name="rgb_topic" value="$(arg rgb_topic)" />
    <arg name="depth_topic" value="$(arg depth_registered_topic)" />
    <arg name="camera_info_topic" value="$(arg camera_info_topic)" />
    <arg name="compressed" value="$(arg compressed)"/>

    <arg name="subscribe_scan" value="$(arg subscribe_scan)"/>
    <arg name="scan_topic" value="$(arg scan_topic)"/>

    <arg name="subscribe_scan_cloud" value="$(arg subscribe_scan_cloud)"/>
    <arg name="scan_cloud_topic" value="$(arg scan_cloud_topic)"/>

    <arg name="visual_odometry" value="$(arg visual_odometry)"/>
    <arg name="icp_odometry" value="$(arg icp_odometry)"/>
    <arg name="odom_topic" value="$(arg odom_topic)"/>
    <arg name="odom_frame_id" value="$(arg odom_frame_id)"/>
    <arg name="odom_args" value="$(arg rtabmap_args)"/>


  </include>

<include file="$(find jhr_navigation)/launch/move_base.launch" />
</launch>
